backend/alert/views.py

```diff
@@ -229,6 +235,249 @@ class RegistrationViewSet(AutoPrefetchViewSetMixin, viewsets.ModelViewSet):
+
+        # getting associated sections from a given section
+        associated_sections = section.associated_sections.all()
+
+        # creating registration group
+        registration_group = RegistrationGroup(section=section)
+
+        # save the registration group
+        registration_group.save()
+
+        # all course sections from the given course
+        # ex. if CIS-120-001, then all_secitons would
+        # all CIS-120 sections
+        all_sections = course.sections.all()
+
+        if "LEC" in types:
+            # to_register will get registered
+            # array of all 'lecture' sections of COURSE
+            to_register = [x for x in all_sections if x.activity == "LEC"]
+            response_codes["LEC"] = self.bulk_register_sections(
+                request=request, sections=to_register, registration_group=registration_group
+            )
+
+        # using this way so that accidental checks are ok on FE
+        if "REC" in types or "LAB" in types:
+            # setting the variable s. depending on rec or lab
+            # however, if lab is passed w class w recs or vice versa
+            # as long as either rec or lec is passed, the "Supplementary"
+            # classes are registered for
+            supplement = "REC"
+
+            if section.activity == "LAB" or (
+                associated_sections and associated_sections[0].activity == "LAB"
+            ):
+                supplement = "LAB"
+
+            # if it is a REC, then need to look at associated lec -> associated rec
+            # needed for examples like MATH 240 / 114
+            if section.activity == "REC" or section.activity == "LAB":
+                associated_recs = associated_sections[0].associated_sections.all()
+                response_codes[supplement] = self.bulk_register_sections(
+                    request=request, sections=associated_recs, registration_group=registration_group
+                )
+            else:
+                response_codes[supplement] = self.bulk_register_sections(
+                    request=request,
+                    sections=associated_sections,
+                    registration_group=registration_group,
+                )
+        # response is
+        return Response(response_codes)
+
+    @action(detail=False, methods=["post"])
+    def multi(self, request):
+        """
+        This is a method that takes a list of registrations in the form
+        of an array and registers for all of them.
+
+        Here is an example of a request.
+        {
+            "courses": ["CIS-120-001", "CIS-120-002"],
+            "auto_resubscribe": true
+        }
+
+        The response is going to be a dictionary that contains has the
+        name of the course as the key and the response message/data as the body.
+        """
+        # creating a list that will hold the response codes
+        response_codes = {}
```

Comment:
This should be an array, right? One per course instead of one per course code. Also will maintain the ordering. Maybe the format could be `[{course: string, response: {...}}]`